src/com/rcv/ResultsWriter.java

```diff
@@ -103,15 +95,61 @@ public ResultsWriter setWinnerToRound(Map<String, Integer> winnerToRound) {
   // function: setElectionConfig
   // purpose: setter for ElectionConfig object
   // param: config the ElectionConfig object to use when writing results
-  public ResultsWriter setElectionConfig(ElectionConfig config) {
+  ResultsWriter setElectionConfig(ElectionConfig config) {
     this.config = config;
     return this;
   }
 
+  // function: generateOverallSummarySpreadsheet
+  // purpose: creates a summary spreadsheet for the full election
+  // param: roundTallies is the round-by-round count of votes per candidate
+  void generateOverallSummarySpreadsheet(
+      Map<Integer, Map<String, BigDecimal>> roundTallies
+  ) {
+    generateSummarySpreadsheet(roundTallies, null, config.visualizerOutput());
+  }
+
+  // function: generatePrecinctSummarySpreadsheet
+  // purpose: creates a summary spreadsheet for the votes in a particular precinct
+  // param: roundTallies is map from precinct to the round-by-round vote count in the precinct
+  void generatePrecinctSummarySpreadsheets(
+      Map<String, Map<Integer, Map<String, BigDecimal>>> precinctRoundTallies
+  ) {
+    // location for output
+    String outputDir = config.precinctOutputDirectory();
+    if (outputDir != null) {
+      // dir is the directory to write these files
+      File dir = new File(config.precinctOutputDirectory());
+      if (!dir.mkdirs()) {
+        Logger.log("Failed to create precinct output dir: " + outputDir);
```

Comment:
Should we throw an exception here?

Reply:
I'd propose validate this in config validate as well.